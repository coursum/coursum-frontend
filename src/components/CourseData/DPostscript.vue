<template>
  <span
    v-if="postscript"
    style="color: #929292"
    class="text-caption text-truncate"
  >
    {{ postscript }}
  </span>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DPostsctipt',
  props: {
    postscript: {
      type: String,
      default: undefined,
    },
  },

});
</script>
