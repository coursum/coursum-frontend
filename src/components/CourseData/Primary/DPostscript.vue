<template>
  <span
    v-if="title && title.postscript && title.postscript[`${curLang}`]"
    style="color: #929292"
    class="text-caption text-truncate"
  >
    {{ title.postscript[`${curLang}`] }}
  </span>
</template>

<script lang="ts">
import Vue from 'vue';
import { Title } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DPostsctipt',
  props: {
    idx: {
      type: Number,
      default: 0,
    },
  },

  computed: {
    title(): Title {
      return this.$store.state.courseDatas[this.idx].title;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>
