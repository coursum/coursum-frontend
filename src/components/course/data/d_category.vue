<template>
  <div
    class="px-6 text-truncate"
  >
    {{ categoryData ? categoryData : $t("others") }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DCategory',
  props: {
    category: { type: Object, default: basicTemplate },
  },
  computed: {
    categoryData(): string | null | undefined {
      return this.category?.[this.curLang];
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
div{
  font-size: 0.7rem;
  color: gray;
}
</style>

<i18n>
{
  "en": {
    "others": "others"
  },
  "jp": {
    "others": "その他"
  }
}
</i18n>
