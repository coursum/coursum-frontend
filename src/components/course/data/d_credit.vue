<template>
  <v-chip
    v-if="credit"
    style="color: #929292"
    outlined
    x-small
  >
    {{ credit }}{{ $t("credit") }}
  </v-chip>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DCredit',
  props: {
    credit: {
      type: Number,
      default: undefined,
    },
  },
});
</script>

<i18n>
{
  "en": {
    "credit": "credit"
  },
  "jp": {
    "credit": "単位"
  }
}
</i18n>
