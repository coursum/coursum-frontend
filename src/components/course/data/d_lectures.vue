<template>
  <div
    v-if="lecturers && 0 < lecturers.length"
    :class="{'text-truncate': textTruncate}"
    class="px-6"
  >
    <template
      v-for="(lecturer, i) in lecturers"
    >
      <span
        v-if="lecturer.name && lecturer.name[`${curLang}`]"
        :key="i"
        style="display:inline-block;"
        class="mr-2"
      >
        {{ lecturer.name[`${curLang}`] }}
      </span>
    </template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DLectures',
  props: {
    lecturers: {
      type: Array,
      default() {
        return undefined;
      },
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
span{
  font-size: 0.85rem;
  color: var(--v-wordcolor-base);
}
</style>
