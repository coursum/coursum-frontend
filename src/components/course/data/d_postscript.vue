<template>
  <span
    v-if="postscriptData"
    style="color: #929292"
    class="px-6 text-caption text-truncate"
  >
    {{ postscriptData }}
  </span>
</template>

<script lang="ts">
import { PropType, computed, defineComponent } from '@vue/composition-api';

import { Basic, basicTemplate } from '@/assets/CourseInfo';

export default defineComponent({
  name: 'DPostsctipt',
  props: {
    postscript: {
      type: Object as PropType<Basic>,
      default: basicTemplate,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'jp' | 'en');

    const postscriptData = computed((): string | null | undefined => props
      .postscript?.[curLang.value]);

    return {
      postscriptData,
    };
  },
});
</script>
