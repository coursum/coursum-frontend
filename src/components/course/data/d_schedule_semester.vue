<template>
  <v-chip v-if="semesterData"
          color="caption"
          outlined
          x-small
  >
    {{ semesterData }}
  </v-chip>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleSemester',
  props: {
    semester: {
      type: Object as PropType<Course['schedule']['semester']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');
    const semesterData = computed(() => props.semester[curLang.value]);

    return {
      semesterData,
    };
  },
});
</script>
