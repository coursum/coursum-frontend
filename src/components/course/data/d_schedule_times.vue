<template>
  <span v-if="courseTimes">
    <v-chip v-for="(time, i) in courseTimes" :key="i"
            color="caption"
            outlined
            x-small
    >
      {{ time }}
    </v-chip>
  </span>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleTimes',
  props: {
    times: {
      type: Object as PropType<Course['schedule']['times']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');

    const courseTimes = computed(() => {
      const times = props.times[curLang.value] || '';

      return times.split(',').map((time) => time.trim());
    });

    return {
      courseTimes,
    };
  },
});
</script>
