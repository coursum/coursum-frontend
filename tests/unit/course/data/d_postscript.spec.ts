import { shallowMount } from '@vue/test-utils';

import { titlePostscriptMock } from '../../../data';
import { i18n } from '../../../setup';

import DPostscript from '@/components/course/data/d_postscript.vue';

const correct = titlePostscriptMock.en;

const options = {
  i18n,
  propsData: {
    postscript: titlePostscriptMock,
  },
  mocks: {
    $t: (msg: string) => msg,
  },
};

describe('DPostscript', () => {
  it('should render correct markup when prop is correct', () => {
    options.propsData.postscript.en = correct;
    const wrapper = shallowMount(DPostscript, options);

    const span = wrapper.find('span');
    expect(span.text()).toEqual(correct);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render anything when prop is falsy', () => {
    [undefined, null, ''].forEach((falsyProp: undefined| null | string) => {
      options.propsData.postscript.en = falsyProp;
      const wrapper = shallowMount(DPostscript, options);

      const span = wrapper.find('span');
      expect(span.exists()).toBe(false);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
