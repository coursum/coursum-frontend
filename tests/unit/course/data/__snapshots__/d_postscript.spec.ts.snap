// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DPostscript should not render anything when prop is falsy 1`] = `<!---->`;

exports[`DPostscript should not render anything when prop is falsy 2`] = `<!---->`;

exports[`DPostscript should not render anything when prop is falsy 3`] = `<!---->`;

exports[`DPostscript should render correct markup when prop is correct 1`] = `
<span
  class="px-6 text-caption text-truncate"
  style="color: rgb(146, 146, 146);"
>
  
  postscript en

</span>
`;
